src/Vision/testpi.py

```diff
@@ -0,0 +1,378 @@
+            lum: A list of two numbers the are the min and max luminance.
+        Returns:
+            A black and white numpy.ndarray.
+        """
+        out = cv2.cvtColor(input, cv2.COLOR_BGR2HLS)
+        return cv2.inRange(out, (hue[0], lum[0], sat[0]), (hue[1], lum[1], sat[1]))
+
+    @staticmethod
+    def __find_contours(input, external_only):
+        """Sets the values of pixels in a binary image to their distance to the nearest black pixel.
+        Args:
+            input: A numpy.ndarray.
+            external_only: A boolean. If true only external contours are found.
+        Return:
+            A list of numpy.ndarray where each one represents a contour.
+        """
+        if(external_only):
+            mode = cv2.RETR_EXTERNAL
+        else:
+            mode = cv2.RETR_LIST
+        method = cv2.CHAIN_APPROX_SIMPLE
+        im2, contours, hierarchy = cv2.findContours(input, mode=mode, method=method)
+        return contours
+
+
+configFile = "/boot/frc.json"
+
+
+class CameraConfig: pass
+
+
+rightAngle = -15
+leftAngle = -75
+deadzone = 15
+
+team = None
+server = False
+width = None
+height = None
+imageMidpoint = None
+
+cameraConfigs = []
+
+"""Report parse error."""
+
+
+def parseError(str):
+    print("config error in '" + configFile + "': " + str, file=sys.stderr)
+
+
+"""Read single camera configuration."""
+
+
+def readCameraConfig(config):
+    cam = CameraConfig()
+    global width
+    global height
+    global imageMidpoint
+
+    # name
+    try:
+        cam.name = config["name"]
+    except KeyError:
+        parseError("could not read camera name")
+        return False
+
+    # path
+    try:
+        cam.path = config["path"]
+    except KeyError:
+        parseError("camera '{}': could not read path".format(cam.name))
+        return False
+
+    try:
+        width = config["width"]
+        imageMidpoint = (width / 2)
+    except KeyError:
+        parseError("could not read width")
+        return False
+
+    try:
+        height = config["height"]
+    except KeyError:
+        parseError("could not read height")
+        return False
+
+    # stream properties
+    cam.streamConfig = config.get("stream")
+
+    cam.config = config
+
+    cameraConfigs.append(cam)
+    return True
+
+
+"""Read configuration file."""
+
+
+def readConfig():
+    global team
+    global server
+
+    # parse file
+    try:
+        with open(configFile, "rt") as f:
+            j = json.load(f)
+    except OSError as err:
+        print("could not open '{}': {}".format(configFile, err), file=sys.stderr)
+        return False
+
+    # top level must be an object
+    if not isinstance(j, dict):
+        parseError("must be JSON object")
+        return False
+
+    # team number
+    try:
+        team = j["team"]
+    except KeyError:
+        parseError("could not read team number")
+        return False
+
+    # ntmode (optional)
+    if "ntmode" in j:
+        str = j["ntmode"]
```

Comment:
Don't use the name `str`, that's the name of the string class...

Reply:
this is auto code, if you change it it turns white instead. Does it want to be white? (Not trying to joke btw)

Reply:
Yes it does - the reason it shows up as a special color is because `str` is a built in name, it's better to use something else instead

Reply:
Ben told me to change it to "mode" after consultation